demos/taco-demo/src/App.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  conditions,
+  decrypt,
+  encrypt,
+  getPorterUri,
+  initialize,
+  ThresholdMessageKit,
+} from '@nucypher/taco';
+import { Mumbai, useEthers } from '@usedapp/core';
+import { ethers } from 'ethers';
+import React, { useEffect, useState } from 'react';
+
+import { ConditionBuilder } from './ConditionBuilder';
+import { Decrypt } from './Decrypt';
+import { Encrypt } from './Encrypt';
+import { Spinner } from './Spinner';
+
+export default function App() {
+  const { activateBrowserWallet, deactivate, account, switchNetwork, chainId } =
+    useEthers();
+
+  const [loading, setLoading] = useState(false);
+  const [condition, setCondition] = useState<conditions.Condition>();
+  const [encryptedMessage, setEncryptedMessage] =
+    useState<ThresholdMessageKit>();
+  const [decryptedMessage, setDecryptedMessage] = useState<string>();
+  const [decryptionErrors, setDecryptionErrors] = useState<string[]>([]);
+
+  useEffect(() => {
+    initialize();
+    switchNetwork(Mumbai.chainId);
+  }, []);
+
+  const encryptMessage = async (message: string) => {
+    if (!condition) {
+      return;
+    }
```

Comment:
it's not obvious to me what the intended flow control is here, are you using `condition` as some sort of status indication?  Why do you return nothing here instead of raising an error?

Reply:
`condition` could be `undefined` in this scope, but that should never happen. This check is for type safety: `tsc` will coalesce `Condition | undefined` type into `undefined` in this scope. I don't think this short demo needs to implement a red path with error messages, validation, etc.